src/workerd/io/worker.c++

```diff
@@ -1145,6 +1145,96 @@ Worker::Isolate::Isolate(kj::Own<Api> apiParam,
+    if (features.getHandleCrossRequestPromiseResolution()) {
+      lock->v8Isolate->SetPromiseCrossContextResolveCallback(
+          [](v8::Isolate* isolate, v8::Local<v8::Value> tag, v8::Local<v8::Data> reactions,
+              v8::Local<v8::Value> argument,
+              std::function<void(v8::Isolate * isolate, v8::Local<v8::Data> reactions,
+                  v8::Local<v8::Value> argument)> callback) {
+        try {
+          auto& js = jsg::Lock::from(isolate);
+          // The promise tag is generally opaque except for right here. The tag
+          // wraps an instanceof kj::Own<IoCrossContextExecutor>, which wraps an atomically
+          // refcounted pointer to the DeleteQueue for the correct isolate.
+          // We simply pass the given callback, reactions, and argument to
+          // a function that will be added to the queue inside DeleteQueue.
+          // The next time the relevant IoContext is entered, this queue will
+          // be drained and the actions will be run. Adding the task to the
+          // delete queue will also signal the IoContext that it should wake
+          // up and drain the queue. Simple, eh?
+          //
+          // A word of warning tho! It is possible for the IoContext to be
+          // destroyed before the promise is resolved. Any actions that have
+          // already been added to the queue would end up being dropped silently
+          // on the floor. Actions that are added to the queue now will be run
+          // immediately in the wrong IoContext.
+          auto& ref = jsg::unwrapOpaqueRef<kj::Own<IoCrossContextExecutor>>(isolate, tag);
+          ref->execute(js,
+              [reactions = v8::Global<v8::Data>(isolate, reactions),
+                  argument = jsg::JsRef(js, jsg::JsValue(argument)),
+                  callback = kj::mv(callback)](jsg::Lock& js) mutable {
+            try {
+              callback(js.v8Isolate, reactions.Get(js.v8Isolate), argument.getHandle(js));
+            } catch (jsg::JsExceptionThrown&) {
+              // Exceptions here are generally unexpected but possible. Let's just
+              // return control back to V8 which should handle the case.
+            } catch (...) {
+              auto ex = kj::getCaughtExceptionAsKj();
+              KJ_LOG(ERROR, "Scheduling deferred promise continuations failed unexpectedly", ex);
+              jsg::throwInternalError(js.v8Isolate, kj::mv(ex));
+            }
+          });
+        } catch (jsg::JsExceptionThrown&) {
+          // Exceptions here are generally unexpected but possible because the jsg::Promise
+          // then can fail if the isolate is in the process of being torn down. Let's just
+          // return control back to V8 which should handle the case.
+        } catch (...) {
+          auto ex = kj::getCaughtExceptionAsKj();
+          KJ_LOG(ERROR, "Scheduling deferred promise continuations failed unexpectedly", ex);
```

Comment:
I think the exception will get logged twice since `throwInternalError()` will also log it?

(Same issue above on line 1223.)